pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarConnectionFactory.java

```
@@ -1010,14 +1055,9 @@ public void close() {
       }
     }
 
-    for (Producer<?> producer : producers.values()) {
-      try {
-        producer.close();
-      } catch (PulsarClientException ignore) {
-        // ignore
-        Utils.handleException(ignore);
-      }
-    }
+    // Invalidate all producers so the removal listener closes them
+    producers.invalidateAll();
```

how does this deal with exceptions ? are they swallowed ?

---

there are no checked exceptions on the method. The exception due to our custom removal listener, gets swallowed.